File: trendify.Client/src/app/models/order/orderDetails-model.ts
import { DeliveryAddressModel } from "./deliveryAddress-model";
import { OrderItemModel } from "./orderItem-model";

export class OrderDetailsModel {
    id!: number;
    orderNumber!: string;
    orderDate!: string;
    fullName!: string;
    email!: string;
    phoneNumber!: string;
    status!: string;
    deliveryAddress!: DeliveryAddressModel;
    items: OrderItemModel[] = [];
    total!: number;

    constructor(init?: Partial<OrderDetailsModel>) {
        if (init) {
            Object.assign(this, init);

            if (init.deliveryAddress) {
                this.deliveryAddress = new DeliveryAddressModel(init.deliveryAddress);
            }
            if (init.deliveryAddress) {
                this.deliveryAddress = new DeliveryAddressModel(init.deliveryAddress);
            }
            if (Array.isArray(init.items)) {
                this.items = init.items.map(i => new OrderItemModel(i));
            }
        }
    }
}

File: trendify.Server/trendify.Core/Models/Orders/OrderDetailsModel.cs
namespace trendify.Core.Models.Orders
{
    public class OrderDetailsModel
    {
        public int Id { get; set; }
        public string OrderNumber { get; set; } = null!;
        public DateTimeOffset OrderDate { get; set; }

        public string FullName { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;

        public string Status { get; set; } = null!;
        public DeliveryAddressModel DeliveryAddress { get; set; } = null!;

        public List<OrderItemModel> Items { get; set; } = new List<OrderItemModel>();
        public decimal Total { get; set; }
    }
}
